src/lib.rs

```diff
@@ -46,17 +48,27 @@
 /// # use copy_in_place::copy_in_place;
 /// let mut bytes = *b"Hello, World!";
 ///
-/// copy_in_place(&mut bytes, 1, 8, 4);
+/// copy_in_place(&mut bytes, 1..5, 8);
 ///
 /// assert_eq!(&bytes, b"Hello, Wello!");
 /// ```
-pub fn copy_in_place<T: Copy>(slice: &mut [T], src: usize, dest: usize, count: usize) {
-    // Avoid computing `src + count` or `dest + count`, which could overflow.
-    assert!(count <= slice.len());
-    assert!(src <= slice.len() - count);
-    assert!(dest <= slice.len() - count);
+pub fn copy_in_place<T: Copy, R: RangeBounds<usize>>(slice: &mut [T], src: R, dest: usize) {
+    let src_start = match src.start_bound() {
+        Bound::Included(&n) => n,
+        Bound::Excluded(&n) => n.saturating_add(1),
+        Bound::Unbounded => 0,
+    };
+    let src_end = match src.end_bound() {
+        Bound::Included(&n) => n.saturating_add(1),
```

Comment:
I wonder if this is technically correct. If a slice actually ends at the largest possible address (I think that's one less than the maximum representable, to allow C to compute the `p + len` pointer?), then an inclusive range that passes the end of the slice by one would be converted from the panic it should be to a valid slice one element shorter than the caller asked for.